### ISSUE TITLE
Is there a way to retrieve the filename of cmake.launchTargetPath

### ISSUE THREAD

**Issue by PetePriority** [microsoft/vscode-cmake-tools#632]
### Brief Issue Summary

My launch configuration uses remote gdb debugging using pipeTransport like so:
```
   {
        "version": "0.2.0",
        "configurations": [
            {
                "name": "(gdb) Pipe Launch",
                "type": "cppdbg",
                "request": "launch",
                "program": "${command:cmake.launchTargetPath}",
                "cwd": "${workspaceFolder}",
                "preLaunchTask": "Deploy",
                "pipeTransport": {
                    "pipeCwd": "${workspaceRoot}",
                    "debuggerPath": "/usr/bin/gdb",
                    "pipeProgram": "/usr/bin/ssh",
                    "pipeArgs": [
                        "root@target"
                    ],
                },
                "MIMode": "gdb",
                "setupCommands": [
                    {
                        "description": "Enable pretty-printing for gdb",
                        "text": "-enable-pretty-printing",
                        "ignoreFailures": true
                    },
                ],
            }
        ]
    }
```
This works fine, as long as the deployment task puts the executable returned by `cmake.launchTargetPath` on the target at the same location as on the local machine.
Is there a way to extract the filename part of `cmake.launchTargetPath` such that I can have custom deployment locations? C.f. https://github.com/Microsoft/vscode-cpptools/issues/2110#issuecomment-396731501 

### Platform and Versions

- **Operating System**: Debian 9.2
- **CMake Version**: 3.7.2
- **VSCode Version**: 1.30.1
- **CMake Tools Extension Version**: 1.1.3
- **Compiler/Toolchain**: Yocto/toradex toolchain for ARM

**Comment by jclay** [comment]
It looks like you could fetch the `launchTarget` (https://github.com/vector-of-bool/vscode-cmake-tools/blob/6a0218b67119ce3f9101f041182392fafa4e12ab/src/api.ts#L275)

Could you instead have program call a script which accepts the target and then builds the path you need, and then calls the program?

**Comment by PetePriority** [comment]
Using an external script would be possible, but I would like to keep the configuration as simple as possible.

How would I access the launchTarget value from within the launch json? I've already tried `"program": "${command:cmake.launchTarget}",` but that didn't work.

**Comment by KoeMai** [comment]
The Implementation does not support it yet. I took a looked into the implementation, the information is very close to the extension API `api.ExecutableTarget`.

**Comment by minkir014** [comment]
You said the deployment made the matter works well. So what is the problem?

**Comment by bobbrow** [comment]
This is available in [1.4.2](https://github.com/microsoft/vscode-cmake-tools/releases/tag/1.4.2)
